Assets/Code/Scripts/NoclipRealityManagement/NoclipMovement.cs

```diff
@@ -47,61 +61,161 @@ public class NoclipMovement : MonoBehaviour
-            float currentSpeed = _movementSpeed;
-            
             _horizontalInput = Input.GetAxisRaw("Horizontal");
             _verticalInput = Input.GetAxisRaw("Vertical");
+            bool moveUp = Input.GetButton("NoclipMoveUp");
+            bool moveDown = Input.GetButton("NoclipMoveDown");
+            bool boost = Input.GetButton("BoostNoclip");
+            _upDownInput = 0;
+            if(moveUp){
+                _upDownInput += 1;
+            }
+            if(moveDown){
+                _upDownInput += -1;
+            }
 
-            deltaPosition += _transform.forward * _verticalInput + _transform.right * _horizontalInput;
+            if (boost)
+            {
+                _maxSpeed = _boostedSpeed;
+                _acceleration = _boostAcceleration;
+            }
+            else
+            {
+                _maxSpeed = _baseSpeed;
+                _acceleration = _baseAcceleration;
+            }
 
             
-            if (Input.GetButton("BoostNoclip"))
-                currentSpeed = _boostedSpeed;
-               
-            /*if (Input.GetKey(KeyCode.W))
-                deltaPosition += _transform.forward;
+            //_horizontalInput and _verticalInput move with linear acceleration in the direction of the camera
+            if (_horizontalInput != 0 || _verticalInput != 0 || _upDownInput != 0)
+            {
+                //increase speed in direction of camera with acceleration
+                _speed += (_transform.forward * _verticalInput).normalized * _acceleration * Time.deltaTime;
+                _speed += (_transform.right * _horizontalInput).normalized * _acceleration * Time.deltaTime;
+                _speed += (_transform.up * _upDownInput).normalized * _acceleration * Time.deltaTime;
+                //limit speed
+                _speed = _speed.normalized * Mathf.Min(_speed.magnitude, _maxSpeed);
+            }
+            else
+            {
+                Vector3 deltaSpeed = _speed.normalized * _acceleration * Time.deltaTime;
+                if(_speed.magnitude > deltaSpeed.magnitude && _speed.magnitude > _maxSpeed * _smoothBrakeRange)
+                {
+                    _speed -= deltaSpeed;
+                    _smoothBrake = false;
+                }
+                else
+                {
+                    _smoothBrake = true;
+                }
+            }
+            //move in direction of speed
+            transform.position += _speed * Time.deltaTime;
+        }
+        else
+        {
+            _speed = Vector3.zero;
+        }
+    }
 
-            if (Input.GetKey(KeyCode.S))
-                deltaPosition -= _transform.forward;
+    private bool _smoothBrake = false;
+
+    private void FixedUpdate(){
+        if(_smoothBrake){
+            _speed *= _smoothBrakeFactor;
+        }
+    }*/
 
-            if (Input.GetKey(KeyCode.A))
-                deltaPosition -= _transform.right;
+void FixedUpdate()
```

Comment:
this implies 50fps movement animation, Update with proper deltatime should be used instead